### DISCUSSION TITLE
Any way to set the WasmShellOutputPackagePath

### DISCUSSION THREAD

**Discussion by groogiam** [unoplatform/Uno.Wasm.Bootstrap#560] [Q&A] [answered]
Is there any way to set the output package path to a fixed folder instead of package_{random_string}? I'm not sure of the purpose of the random naming but it makes it very difficult to package the output into a larger web site especially because the uno-config.js references all the files by this random path.  I would like to be able to just have the output called package.

**Comment by jeromelaban** [comment, marked as answer]
At this time, there isn't. The path is not random, and is built based on the contents of the dist folder. The reason for this is to ensure that caches don't break the runtime, as all files need to be exactly in sync when downloaded.

In many occasions, without a unique path, intermediate caches (explicit or transparent proxies, browsers, CDNs, etc...) would unevenly cache `.clr` files, causing apps to break.

**Reply by groogiam** [reply]
Thanks for the quick response. Any plans to make this configurable? I understand the reasoning but seems a little heavy handed. I don't see other wasm frameworks like Blazor doing this.

**Reply by jeromelaban** [reply]
I don't think we'll include such a feature, no. Can you explain in more details why this is needed? In theory, you should not be needing to interact with the contents of the dist folder directly.

**Reply by groogiam** [reply]
I'm trying to use Uno.Wasm.Bootstrapper to build an offline ASP.NET core server that runs in a service worker so I'm not using the generated index.html. The directory changing every time makes it much more difficult to integrated the output into parent application that requires the offline functionality.

**Reply by jeromelaban** [reply]
I see. The only mode would be closed to what you're doing is: https://github.com/unoplatform/Uno.Wasm.Bootstrap/blob/main/doc/features-embedded.mode.md.

It is broken in the current 7.0 builds, but works in 3.3 builds.

**Reply by groogiam** [reply]
Thanks. Looks like the generated embedded.js has dependencies on the document so it will not work in a service worker. I think I've found a work around using some msbuild magic to generate a custom script to load the required bootstrapper javascript files into the service worker.
